nexis/lib/pages/home.dart

```diff
@@ -11,83 +19,256 @@ class Home extends StatefulWidget {
         child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            buildMessagesListView(),
+            buildMessageInputArea(),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildMessagesListView() {
+    return Expanded(
+      child: messages.isEmpty && messagesLoaded
+          ? const Center(
+              child: Text(
+                'No messages',
+                style: TextStyle(color: Colors.white),
+              ),
+            )
+          : ListView.builder(
+              padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 24),
+              shrinkWrap: true,
+              itemCount: messages.length,
+              itemBuilder: (context, index) => buildMessageItem(index),
+            ),
+    );
+  }
+
+  Widget buildMessageItem(int index) {
+    final message = messages[index];
+    final sender = message.sender;
+    final timestamp = message.timestamp.toDate();
+    final content = message.content;
+
+    final formattedTimestamp = TimeFormat.formattedTimestamp(timestamp);
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.start,
           children: [
             Text(
-              'Nexis',
-              style: Theme.of(context).textTheme.displayMedium,
+              sender,
+              style: const TextStyle(
+                color: Colors.white,
+                fontWeight: FontWeight.bold,
+              ),
             ),
-            const SizedBox(height: 10),
-            // TODO: Replace this in the future with a custom text field
-            SizedBox(
-              width: MediaQuery.of(context).size.width * 0.6,
-              child: TextField(
-                controller: messageController,
-                decoration: InputDecoration(
-                  labelText: 'Enter your message',
-                  fillColor: Theme.of(context).colorScheme.tertiary,
-                  filled: true,
-                  border: const OutlineInputBorder(
-                    borderSide: BorderSide(color: Colors.white),
-                    borderRadius: BorderRadius.all(
-                      Radius.circular(10),
-                    ),
-                  ),
-                  focusedBorder: const OutlineInputBorder(
-                    borderSide: BorderSide(color: Colors.white),
+            const SizedBox(width: 8),
+            Text(
+              formattedTimestamp,
+              style: const TextStyle(
+                color: Colors.white,
+              ),
+            ),
+          ],
+        ),
+        const SizedBox(height: 4),
+        Text(
+          content,
+          style: const TextStyle(
+            color: Colors.white,
+          ),
+        ),
+        const SizedBox(height: 8),
+      ],
+    );
+  }
+
+  Widget buildMessageInputArea() {
```

Comment:
Same with this one

Reply:
this is temporary, Ante is working on this one